src/libUtils/EvmUtils.cpp

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (C) 2022 Zilliqa
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <array>
+#include <iostream>
+#include <mutex>
+#include <string>
+#include <unordered_map>
+
+#include <boost/filesystem.hpp>
+#include "EvmUtils.h"
+
+#include "JsonUtils.h"
+#include "Logger.h"
+#include "common/Constants.h"
+#include "libData/AccountData/Account.h"
+#include "libData/AccountData/TransactionReceipt.h"
+#include "libPersistence/ContractStorage.h"
+#include "libUtils/EvmCallParameters.h"
+#include "libUtils/EvmJsonResponse.h"
+
+using namespace std;
+using namespace boost::multiprecision;
+
+Json::Value EvmUtils::GetEvmCallJson(const EvmCallParameters& params) {
+  Json::Value arr_ret(Json::arrayValue);
+
+  arr_ret.append(params.m_contract);
+  arr_ret.append(params.m_caller);
+  std::string code;
+  try {
+    // take off the EVM prefix
+    std::copy(params.m_code.begin() + 3, params.m_code.end(),
+              std::back_inserter(code));
+    arr_ret.append(code);
+  } catch (std::exception& e) {
+    arr_ret.append(params.m_code);
+  }
+  arr_ret.append(params.m_data);
+  arr_ret.append(params.m_apparent_value.str());
+  arr_ret.append(Json::Value::UInt64(params.m_available_gas));
+
+  if (LOG_SC) {
+    LOG_GENERAL(WARNING, "Sending to EVM-DS" << arr_ret);
+  }
+
+  return arr_ret;
+}
+
+bool EvmUtils::EvmUpdateContractStateAndAccount(
```

Comment:
As noted above, this has to process multiple instructions, not specific to the current contract account. Hence it shouldn't be a parameter.